File: tests/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test/assets/TestClasses/ControllerUpgrade.Fixed.cs
using System.Web.Http;
using System.Web.Mvc;
using System.Collections.Generic;
using Microsoft.AspNetCore.Mvc;

namespace TestProject.TestClasses
{
    [ApiController]
    public partial class ValuesController : ControllerBase
    {
        // GET api/values
        public IEnumerable<string> GetValues()
        {
            return new[] { "value1", "value2" };
        }
    }

    public partial class ValuesController
    {
        // GET api/morevalues
        public IEnumerable<string> GetMoreValues()
        {
            return new[] { "value3", "value4" };
        }
    }

    public class MoviesController : ControllerBase
    {
        // GET api/values
        public IEnumerable<string> GetValues()
        {
            return new[] { "Star Wars", "Iron Man", "Star Trek" };
        }
    }

    public class NotAWebController : Foo.ApiController
    {

    }

    public partial class Controller2
    {
        Bar.Controller Controller;
    }

    public partial class Controller2 : Controller
    {
        public Controller DoSomething(ControllerBase a)
        {
            var x = new List<Controller>();
            return new Controller2();
        }
    }

    public class NotAController : Foo.Controller
    {
        public string Controller { get; set; }

        public void SomeMethod()
        {
            var anonymousType = new { Controller = "A string" };
            if (anonymousType is null)
            {
                var Controller = anonymousType;
                var ControllerBase = new Bar.ApiController { Controller = Controller };
            }

            var x = new Controller();
            var y = Controller.Foo();       // Property access
            var y1 = ControllerBase.Foo();   // Static class access
            var z = new Controller[10];
            ControllerBase x = null;
            var z1 = new Foo.ApiController().Controller;
        }

        public Controller ControllerProperty { get; set; }
        private ControllerBase ControllerField;
    }
}

namespace Foo
{
    public class ApiController
    {
        object Controller
        {
            get;

            set
            {
                var ApiController = string.Empty;
                var x = ApiController.Substring(0, 0);
            }
        }
    }

    public class Controller
    {

    }
}

File: tests/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test/assets/TestClasses/ControllerUpgrade.Fixed.vb
Imports System.Web.Http
Imports System.Web.Mvc
Imports System.Collections.Generic
Imports Microsoft.AspNetCore.Mvc

Namespace TestProject.TestClasses

    Public Class ValuesController2
        Inherits ControllerBase

        ' GET api/values
        Public Function GetValues() As IEnumerable(Of String)
            Dim fooService = New NotAWebController2()

            Return fooService
        End Function
    End Class

    Public Class MoviesController2
        Inherits ControllerBase

        ' GET api/movies
        Public Function GetMovies() As IEnumerable(Of String)
            Return New String() {"Star Wars", "Iron Man", "Star Trek"}
        End Function
    End Class

    Public Class NotAWebController2
        Inherits Foo.ApiController

        Public Function GetValues() As IEnumerable(Of String)
            Return New String() {"value1", "value2"}
        End Function

    End Class

    Public Class ValuesController3
        Inherits Controller

        ' GET api/values
        Public Function GetValues() As IEnumerable(Of String)
            Dim fooService = New NotAWebController2()

            Return fooService
        End Function
    End Class

    Public Class MoviesController3
        Inherits Controller

        ' GET api/movies
        Public Function GetMovies() As IEnumerable(Of String)
            Return New String() {"Star Wars", "Iron Man", "Star Trek"}
        End Function
    End Class

    Public Class NotAWebController3
        Inherits Foo.Controller

        Public Function GetValues() As IEnumerable(Of String)
            Return New String() {"value1", "value2"}
        End Function

    End Class

End Namespace

Namespace Foo
    Public Class ApiController

    End Class

    Public Class Controller

    End Class
End Namespace
